file-sink/src/main/java/org/springframework/cloud/stream/module/file/FileSinkProperties.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.file;
+
+import java.io.File;
+
+import javax.validation.constraints.AssertTrue;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.expression.Expression;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.util.StringUtils;
+
+/**
+ * Properties for the file sink.
+ *
+ * @author Mark Fisher
+ */
+@ConfigurationProperties
+public class FileSinkProperties {
+
+	private static final String DEFAULT_DIR =
+			File.separator + "tmp" +
```

Comment:
I don't think this is sufficient to work on windows, is it?
This will result in `\tmp\dataflow\output`, that is all.

Should somehow do `${java.io.tmpDir} + /dataflow/output` instead